django_advanced_queries/covid_19/tests.py

```diff
@@ -406,19 +420,45 @@ def test_doctor_performed_the_most_medical_examinations(self):
 
     def test_num_of_sick_persons(self):
         with self.assertNumQueries(1):
-            sick_persons = Person.objects.get_sick_persons()
+            results_subquery = MedicalExaminationResult.objects.filter(
+                patient__person=OuterRef('pk')).order_by('-time')
+
+            sick_persons = Person.objects.annotate(
+                last_result=Subquery(
+                    results_subquery.values('result')[:1])).exclude(
+                last_result__in=[MedicalExaminationResult.RESULT_DEAD,
+                                 MedicalExaminationResult.RESULT_HEALTHY])
+
             self.assertEqual(sick_persons.count(), 3)
 
     def test_num_of_sick_hospital_workers(self):
         with self.assertNumQueries(1):
-            sick_hospital_workers = HospitalWorker.objects.get_sick_workers()
+            results_subquery = MedicalExaminationResult.objects.filter(
+                patient__person=OuterRef('person')).order_by('-time')
```

Comment:
it's the same logic as used in `test_num_of_sick_persons`

Reply:
d